main.go

```diff
@@ -301,51 +282,94 @@ func prepareFilter(filters []string) (tracee.Filter, error) {
 			NotEqual: []string{},
 			Enabled:  false,
 		},
+		ContFilter: &tracee.BoolFilter{
+			Set:     false,
+			Enabled: false,
+		},
+		EventsToTrace: []int32{},
 	}
 
+	eventFilter := &tracee.StringFilter{Equal: []string{}, NotEqual: []string{}, Enabled: false}
+	setFilter := &tracee.StringFilter{Equal: []string{}, NotEqual: []string{}, Enabled: false}
+
 	for _, f := range filters {
-		if strings.HasPrefix(f, "uid") {
-			err := parseUintFilter(strings.TrimPrefix(f, "uid"), filter.UIDFilter)
+		filterName := f
+		operatorAndValues := ""
+		operatorIndex := strings.IndexAny(f, "=!<>")
+		if operatorIndex > 0 {
+			filterName = f[0:operatorIndex]
+			operatorAndValues = f[operatorIndex:]
+		}
+
+		// The filters which are more common (container, event, pid, set, uid) can be given using a prefix of them.
+		// Other filters should be given using their full name.
+		// To avoid collisions between filters that share the same prefix, put the filters which should have an exact match first!
+		if filterName == "comm" {
+			err := parseStringFilter(operatorAndValues, filter.CommFilter)
+			if err != nil {
+				return tracee.Filter{}, err
+			}
+			continue
+		}
+
+		if strings.HasPrefix("container", f) || (strings.HasPrefix("!container", f) && len(f) > 1) {
```

Comment:
if `strings.HasPrefix("!container", f)` then for sure `len(f) > 1`, isn't it?

Reply:
No. "!" filter

Reply:
I don't understand... `f` is a string that starts with `!container`, how can it not be longer than 1?

Reply:
It's the other way around. It checks if "f" is a prefix of "container" or "!container", so we can filter with prefix: "--filter c" or "--filter !c"

Reply:
ahh right